libraries/Zigbee/src/ZigbeeCore.cpp

```diff
@@ -98,7 +99,7 @@ static void esp_zb_task(void *pvParameters) {
 
   //NOTE: This is a workaround to make battery powered devices to be discovered as battery powered
   if (((zigbee_role_t)Zigbee.getRole() == ZIGBEE_END_DEVICE) && edBatteryPowered) {
-    zb_set_ed_node_descriptor(0, 0, 0);
+    zb_set_ed_node_descriptor(0, _rx_on_when_idle, 1);
```

Comment:
function is not a class method, so it does not have access to ` _rx_on_when_idle`. You can add a getter to the class and use `Zigbee.getRxOnWhenIdle()` instead

Reply:
Since only one Zigbee exists, another option is to make `_rx_on_when_idle` global `static bool` for the file